core/src/apps/bitcoin/common.py

```diff
@@ -133,6 +133,9 @@ def decode_bech32_address(prefix: str, address: str) -> tuple[int, bytes]:
         raise wire.ProcessError("Invalid address witness program")
     assert witver is not None
     assert raw is not None
+    # check that P2TR address encodes a valid BIP340 public key
+    if witver == 1 and len(raw) == 32 and not bip340.verify_publickey(bytes(raw)):
```

Comment:
Same comment as for T1 (https://github.com/trezor/trezor-firmware/pull/2077/files#r792684749). Either we should skip not check `len(raw) == 32` and let `bip340.verify_publickey()` fail when the length is not 32 bytes or we should check that v1 witness programs are 32 bytes in `bech32.decode()`.

Reply:
https://github.com/trezor/trezor-firmware/pull/2077/commits/46113d6b81155452aa4d4acbbba4d1d1ff61ae8b